### PR TITLE
fixes contents not being removed from the spatial grid when deleted during movement between 2 grid cells

### LINKED ISSUE #75206: Flaky test create_and_destroy: /mob/living/basic/cockroach instance, which is supposed to only be in the contents of a spatial grid cell at coords: (68, 17, 3), was in the contents of 1 spatial grid cell but it was inside the area handled by another grid cell! within the contents of a cell at coords: (68, 34, 3). (code/controllers/subsystem/spatial_gridmap.dm:548)
Flaky tests were detected in [this test run](https://github.com/tgstation/tgstation/actions/runs/4903583179/attempts/1). This means that there was a failure that was cleared when the tests were simply restarted.
Failures:
```
create_and_destroy: [21:04:08] Runtime in stack_trace.dm,4: /mob/living/basic/cockroach instance, which is supposed to only be in the contents of a spatial grid cell at coords: (68, 17, 3), was in the contents of 1 spatial grid cell but it was inside the area handled by another grid cell! within the contents of a cell at coords: (68, 34, 3). (code/controllers/subsystem/spatial_gridmap.dm:548)
proc name:  stack trace (/proc/_stack_trace)
usr: *no key*/(cockroach)
usr.loc: (Lavaland Wastes (66,17,3))
src: null
call stack:
stack trace("/mob/living/basic/cockroach in...", "code/controllers/subsystem/spa...", 548)
Spatial Grid (/datum/controller/subsystem/spatial_grid): untracked movable error(the cockroach (/mob/living/basic/cockroach))
Spatial Grid (/datum/controller/subsystem/spatial_grid): force remove from grid(the cockroach (/mob/living/basic/cockroach))
the cockroach (/mob/living/basic/cockroach): Destroy(0)
the cockroach (/mob/living/basic/cockroach): Destroy(0)
the cockroach (/mob/living/basic/cockroach): Destroy(0)
qdel(the cockroach (/mob/living/basic/cockroach), 0)
the cockroach (/mob/living/basic/cockroach): death(null)
the cockroach (/mob/living/basic/cockroach): death(null)
the cockroach (/mob/living/basic/cockroach): update stat()
...
the lava (66,17,3) (/turf/open/lava/smooth/lava_land_surface): Entered(the cockroach (/mob/living/basic/cockroach), the volcanic floor (66,18,3) (/turf/open/misc/asteroid/basalt/lava_land_surface), null)
the cockroach (/mob/living/basic/cockroach): Move(the lava (66,17,3) (/turf/open/lava/smooth/lava_land_surface), 2, 0, 1)
the cockroach (/mob/living/basic/cockroach): Move(the lava (66,17,3) (/turf/open/lava/smooth/lava_land_surface), 2, 0, 1)
the cockroach (/mob/living/basic/cockroach): Move(the lava (66,17,3) (/turf/open/lava/smooth/lava_land_surface), 2, null)
the cockroach (/mob/living/basic/cockroach): Move(the volcanic floor (67,17,3) (/turf/open/misc/asteroid/basalt/lava_land_surface), 6, 0, 1)
/datum/idle_behavior/idle_rand... (/datum/idle_behavior/idle_random_walk): perform idle behavior(0.1, /datum/ai_controller/basic_con... (/datum/ai_controller/basic_controller/cockroach))
AI Behavior Ticker (/datum/controller/subsystem/processing/ai_behaviors): fire(0)
AI Behavior Ticker (/datum/controller/subsystem/processing/ai_behaviors): ignite(0)
Master (/datum/controller/master): Loop(2)
Master (/datum/controller/master): StartProcessing(0) at stack_trace.dm:4
```

### PR DESCRIPTION
**Pull request by Kylerace**
## About The Pull Request
fixes the flaky test reports for cockroaches being stuck in the spatial grid (which mothblocks seems to have closed all of)

cockroaches get deleted when they die, so theres a spurious unit test failure where if a cockroach is on a tile in grid cell A and moves to a lava tile in grid cell B, they will get killed when lava.Entered() is called, then deleted, and when /atom/movable/Destroy() is called we try to take them out of grid cell B (because their loc is the lava tile) but they were never added to that cell yet because their movement never finished, so that doesnt do anything. THEN moveToNullspace() is called, that movement finishes before the first movement, and then in Moved(old_loc = lava turf) we try to remove it from grid cell B which again doesnt work, and then the first movements Moved(old_loc = original turf) is called where we can actually remove them from the correct grid cell, except we cant because in exit_cell() we subtract `old_target.important_recursive_contents[channel]` from the cells content lists, and since the target is deleted by this point it doesnt have important_recursive_contents. so the fix here is changing this so it subtracts `old_target.important_recursive_contents?[type] || old_target` instead, which works if the target is deleted.

also fixes some Entered() overrides that dont call parent and improves documentation on spatial grid defines
## Why It's Good For The Game
fixes it without needing the change_loc() setter

### CODE REVIEW

File: code/__DEFINES/spatial_gridmap.dm [hunk 1]
```diff
@@ -1,12 +1,13 @@
-///each cell in a spatial_grid is this many turfs in length and width
+/// each cell in a spatial_grid is this many turfs in length and width (with world.max(x or y) being 255, 15 of these fit on each side of a z level)
 #define SPATIAL_GRID_CELLSIZE 17
-///Takes a coordinate, and spits out the spatial grid index (x or y) it's inside
+/// Takes a coordinate, and spits out the spatial grid index (x or y) it's inside
 #define GET_SPATIAL_INDEX(coord) ROUND_UP((coord) / SPATIAL_GRID_CELLSIZE)
-#define GRID_INDEX_TO_COORDS(index) (index * SPATIAL_GRID_CELLSIZE)
+/// changes the cell_(x or y) vars on /datum/spatial_grid_cell to the x or y coordinate on the map for the LOWER LEFT CORNER of the grid cell.
+/// index is from 1 to SPATIAL_GRID_CELLS_PER_SIDE
+#define GRID_INDEX_TO_COORDS(index) ((((index) - 1) * SPATIAL_GRID_CELLSIZE) + 1)
```
Comment (Kylerace) [inline code/__DEFINES/spatial_gridmap.dm:7]: the math was wrong and had an off-by-one error on both x and y axes so it gave the coordinates of the upper right corner of the grid cell, now its lower left

**Comment by Kylerace** [comment]
i forgot that this doesnt actually fix the error message i added because its called in Destroy() before it eventually gets removed :)

**Comment by LemonInTheDark** [comment]
THE PROMISED PR

File: code/__DEFINES/spatial_gridmap.dm (hunk 1 above)
Comment (LemonInTheDark) [reply code/__DEFINES/spatial_gridmap.dm:7]: off by one moment

**Comment by Kylerace** [comment]
ok yeah, i cant think of a better way to fix the unit test error i added than change_loc(). which means that ironically ive fixed the original error but cant fix the error meant to make it easier to diagnose the original error. i dont want to flip back again in this pr so ill just undraft it

File: code/controllers/subsystem/spatial_gridmap.dm [hunk 2]
```diff
@@ -565,7 +565,7 @@ SUBSYSTEM_DEF(spatial_grid)
 
 #ifdef UNIT_TESTS
 	if(untracked_movable_error(to_remove))
-		find_hanging_cell_refs_for_movable(to_remove, remove_from_cells=TRUE)
+		find_hanging_cell_refs_for_movable(to_remove, remove_from_cells=FALSE) //dont remove from cells because we should be able to see 2 errors
```
Comment (Kylerace) [inline code/controllers/subsystem/spatial_gridmap.dm:568]: fun fact: this silenced the actual hard delete error which wasnt a very good decision

**Pull request merged**